site/src/component/QuarterTooltip/Chart.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import { ResponsiveBar, BarTooltipProps, BarDatum } from '@nivo/bar';
+
+import ThemeContext from '../../style/theme-context';
+import { type Theme } from '@nivo/core';
+
+const colors = ['#E8966D', '#60A3D1', '#FFC7DF', '#F5D77F', '#E8966D', '#EBEBEB'];
+
+interface ChartProps {
+  terms: string[];
+}
+
+export default class Chart extends React.Component<ChartProps> {
+  getTheme = (darkMode: boolean): Theme => {
+    return {
+      axis: {
+        ticks: {
+          text: {
+            fill: darkMode ? '#eee' : '#333',
+          },
+        },
+        legend: {
+          text: {
+            fill: darkMode ? '#eee' : '#333',
+          },
+        },
+      },
+    };
+  };
+
+  /*
+   * Create an array of objects to feed into the chart.
+   * @return an array of JSON objects detailing the grades for each class
+   */
+  getTermData = (): BarDatum[] => {
+    let fallCount = 0,
+      winterCount = 0,
+      springCount = 0;
+
+    // for summer, count unique years rather than total terms (e.g. count SS1 2023 and SS2 2023 as one)
+    const summerYears = new Set<string>();
+
+    this.props.terms.forEach((data) => {
+      const year = data.split(' ')[0];
+      const term = data.split(' ')[1];
```

Comment:
i think you could do this with one line:

```suggestion
      const [year, term] = data.split(' ');
```